core-bundle/src/Controller/ContentElement/AbstractContentElementController.php

```diff
@@ -30,8 +30,35 @@ public function __invoke(Request $request, ContentModel $model, string $section,
         $this->addSectionToTemplate($template, $section);
         $this->tagResponse(['contao.db.tl_content.'.$model->id]);
 
-        return $this->getResponse($template, $model, $request);
+        $response = $this->getResponse($template, $model, $request);
```

Comment:
I think we should add a check here to see if `$response->headers->getCacheControlDirective('public')`. Otherwise we automatically turn `private` responses into public ones, wdyt?